lib/three_scale/middleware/presigned_downloads.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  module Middleware
+    class PresignedDownloads
+      RE_SEPARATOR = ::File::SEPARATOR == "/" ? "/" : "[#{Regexp.escape(::File::SEPARATOR)}/]"
+
+      delegate *%i[requires_signing? good_signature? signature_at to_time], to: :class, private: true
```

Comment:
```suggestion
      delegate :requires_signing?, :good_signature?, :signature_at, :to_time, to: :class, private: true
```

I consider the suggested syntax cleaner and less confusing. Also, RuboCop (with 3.1 Ruby target version) was complaining:
```
RuboCop: Ambiguous splat operator. Parenthesize the method arguments if it's surely a splat operator, or add a whitespace to the right of the `*` if it should be a multiplication. [Lint/ AmbiguousOperator]
```

and suggesting to wrap all args in `()` which makes the expression look even more cumbersome.

Reply:
I always thought it complained when arrays can be written in this way :angry:

Reply:
Also, just wondering - how do you decide whether the method should be a class method or an instance method? Looks like all private methods in this class can be class methods?...

Reply:
The middleware is instantiated. So it is more direct to call instance methods. Other than that - true. All can be class methods.

At the moment methods that are only called by the middleware itself, are instance protected methods. While methods that are called by the paperclip integration are class methods so they are easily accessible to it.